src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -381,11 +381,14 @@ add_axioms_for_char_literal(const function_application_exprt &f)
   // for C programs argument to char literal should be one string constant
   // of size 1.
   if(
-    arg.operands().size() == 1 && arg.op0().operands().size() == 1 &&
-    arg.op0().op0().operands().size() == 2 &&
-    arg.op0().op0().op0().id() == ID_string_constant)
+    arg.operands().size() == 1 &&
+    to_unary_expr(arg).op().operands().size() == 1 &&
```

Comment:
Any idea what this is expecting? The chain of `op0`s without checking *what* unary operation it's looking at seems suspicious

Reply:
I think it's an equality whose lhs is an address_of string_constant -- @romainbrenguier ?

Reply:
this could have been an `address_of` although I can't be sure now. This is most likely dead code, as this is for C programs, for which the solver is not tested currently, and wouldn't work without some clean-up.